runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -77,14 +77,24 @@ local TSCallbackNames = {
 ---@field private _opts table Options
 ---@field private _parser TSParser Parser for language
 ---@field private _has_regions boolean
----@field private _regions table<integer, Range6[]>?
+---
 ---List of regions this tree should manage and parse. If nil then regions are
 ---taken from _trees. This is mostly a short-lived cache for included_regions()
+---@field private _regions table<integer, Range6[]>?
+---
+---Inverse of `_regions`. start byte of range 1 ↦ { region1, index1, region2, index2, .. }.
+---Used for checking if a new region is already managed by this parser, so that it can be parsed
+---incrementally.
+---@field private _regions_inv table<integer, (Range6[]|integer)[]>?
```

Comment:
I find this field quite confusing.

Would it be easier to split this into two table, one for the regions and one for the indexes?